### PR TITLE
slurmd: add singularity-install action

### PR DESCRIPTION
**Pull request by jaimesouza**
**Please provide description of the purpose of this pull request, as well as its
motivation and context.**

Add an action on charm-slurmd to install singularity using the
resources singularity-deb and singularity-rpm, both also included
in this commit.

The .deb (singularity-ce_3.10.2-focal_amd64.deb) and
.rpm (singularity-ce-3.10.2-1.el7.x86_64.rpm) file should be
downloaded from Singulary Github Releases.

Note: The charm-slurmd/requirements.txt must be updated to use
the version of slurm-ops-manager with the Singulary class
implemented (https://github.com/omnivector-solutions/slurm-ops-manager/pull/81) , before applying the action.

**How was the code tested?**

A simple test (80-singularity.bats) was implemented to verify if the singularity is installed. 
It was tested with Ubuntu and Centos7 systems.

Final checklist
- [X] I am the author of these changes, or I have the rights to submit them.
- [ ] I added the relevant changed to the README and/or documentation.
- [X] I self reviewed my own code.
- [x] I updated the `CHANGELOG` according to the [contributing
  guide](https://omnivector-solutions.github.io/osd-documentation/master/contributing.html#changelog)

### CODE REVIEW

File: tests/80-singularity.bats [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/npx bats
+
+load "../node_modules/bats-support/load"
+load "../node_modules/bats-assert/load"
+. "tests/utils.sh"
+
+
+@test "test singularity is installed" {
+	partition="Unit-test-partition"
+	myjuju config --model $JUJU_MODEL slurmd partition-name="$partition"
```
Comment (heitorPB) [inline tests/80-singularity.bats]: This is not needed

```suggestion
```

File: charm-slurmd/src/charm.py [hunk 2]
```diff
@@ -501,6 +503,22 @@ def nvidia_install(self, event):
...
+        if resource_name != "":
+            logger.debug(f"#### Retrieving singularity resource: {resource_name}")
+
+            try:
+                resource_path = self.model.resources.fetch(resource_name)
+                logger.debug(f"#### Found singularity resource: {resource_path}")
+                self._slurm_manager.singularity.install(resource_path)
+                event.set_results({'installation': 'Successfull.'})
+            except ModelError as e:
+                logger.error(f"## Missing singularity resource - {e}")
```
Comment (heitorPB) [inline charm-slurmd/src/charm.py:520]:
```suggestion
                logger.error(f"## Missing singularity resource - {e}")
                event.fail(message=f'Error installing Singularity: {e.output}')
```

File: charm-slurmd/actions.yaml [hunk 3]
```diff
@@ -88,3 +88,12 @@ nvidia-install:
...
     Nvidia repository as the default and install the latest drivers from it.
+
+singularity-install:
+  description: >
+    Install Singularity. This might take a few minutes to complete.
+
+    This action will install singularity using the officials .deb (Ubuntu) 
+    or .rpm (Centos) packages retrieved from GitHub Releases.
+
+    Note: The .deb or .rpm files must be stored as juju resources.
```
Comment (heitorPB) [inline charm-slurmd/actions.yaml]:
```suggestion
    This action will install singularity using the official .deb (Ubuntu) 
    or .rpm (CentOS) packages retrieved from GitHub Releases.

    Note: The .deb or .rpm files must be suppliead as Juju resources.
```

**Review by heitorPB — CHANGES_REQUESTED** [review]
Nice work! Left some minor comments in here.

File: charm-slurmd/requirements.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
 ops==1.3.0
 etcd3gw==1.0.2
-git+https://github.com/omnivector-solutions/slurm-ops-manager.git@0.8.4
+git+https://github.com/omnivector-solutions/slurm-ops-manager.git@0.8.4
```
Comment (heitorPB) [inline charm-slurmd/requirements.txt]:
```suggestion
git+https://github.com/omnivector-solutions/slurm-ops-manager.git@0.8.5
```

Comment (heitorPB) [reply charm-slurmd/requirements.txt]: Could you update all other charms' requirements.txt as well?

Comment (jaimesouza) [reply charm-slurmd/requirements.txt]: Yes, I gonna do that.

**Comment by heitorPB** [comment]
Nice work!

It is missing an entry in the changelog. Also, when can we have the docs for this PR?

**Comment by jaimesouza** [comment]
> Nice work!
> 
> It is missing an entry in the changelog. Also, when can we have the docs for this PR?

@heitorPB, I included the entry in changelog. I am gonna wait your confirmation to merge. After I finish some tasks, I gonna update the docs for this PR.

File: CHANGELOG [hunk 5]
```diff
@@ -7,6 +7,11 @@ This file keeps track of all notable changes to the Slurm Charms.
 Unreleased
 ----------
 
+0.9.3 - 2022-08-18
+------------------
+
```
Comment (heitorPB) [inline CHANGELOG]: Adding an entry to the Changelog should not change the version of the system. Developing a new feature and carving out a new release are two different processes, let's not mix them together.

```suggestion
```

Comment (jaimesouza) [reply CHANGELOG]: Ok

**Comment by heitorPB** [comment]
> After I finish some tasks, I gonna update the docs for this PR.

Do you have an ETA for that? Can we expect it this week?

**Review by heitorPB — APPROVED** [review]
Nice work!

**Comment by jaimesouza** [comment]
> > After I finish some tasks, I gonna update the docs for this PR.
> 
> Do you have an ETA for that? Can we expect it this week?

I think I am gonna finish the in progress (and to do) tasks until tomorrow. So I'd say I could work on the docs tomorrow as well. Can you point me to the repo/file I should change to update the docs?

**Comment by jaimesouza** [comment]
Also, can I squash and merge already?

**Pull request merged**